extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -570,7 +580,12 @@ private static Table readTableInternal(
                 return readPartitionedTableWithMetadata(source, instructions);
             }
             final Path firstEntryPath;
-            try (final DirectoryStream<Path> sourceStream = Files.newDirectoryStream(sourcePath)) {
+            // Ignore dot files while looking for the first entry
+            try (final DirectoryStream<Path> sourceStream =
+                    Files.newDirectoryStream(sourcePath, (path) -> {
+                        final String filename = path.getFileName().toString();
+                        return !filename.isEmpty() && filename.charAt(0) != '.';
```

Comment:
We could further improve this. The dot filter is good. We should also filter out regular files that don't end in `.parquet`.

Reply:
I didn't add that because we don't want to filter directories not ending with `.parquet`

Reply:
Exactly. I was saying to only filter out *regular files* that don't end in Parquet.

Reply:
Now, that could be much more expensive, since we'd have to stat every file we look at, but since we're only looking for the first that matches our filter it may not be so bad.

Reply:
Discussed offline, skipping this change.